src/helpers/reporthelper.js

```diff
@@ -0,0 +1,127 @@
+const moment = require('moment-timezone');
+const userProfile = require('../models/userProfile');
+
+
+const reporthelper = function () {
+  /**
+   * Get an array of objects, each containing selected fields from
+   * the userProfile collection relative to the weekly summaries.
+   *
+   * @param {integer} startWeekIndex The start week index, eg. 0 for this week.
+   * @param {integer} endWeekIndex The end week index, eg. 1 for last week.
+   */
+  const weeklySummaries = function (startWeekIndex, endWeekIndex) {
+    const pstStartOfWeek = moment().tz('America/Los_Angeles').startOf('week').subtract(startWeekIndex, 'week');
+    const pstEndOfWeek = moment().tz('America/Los_Angeles').endOf('week').subtract(endWeekIndex, 'week');
+    const fromDate = moment(pstStartOfWeek).toDate();
+    const toDate = moment(pstEndOfWeek).toDate();
+
+    return userProfile.aggregate([{
+      $match: {
+        isActive: true,
+      },
+    },
+    {
+      $project: {
+        firstName: 1,
+        lastName: 1,
+        mediaUrl: 1,
+        weeklySummaries: {
+          $filter: {
+            input: '$weeklySummaries',
+            as: 'ws',
+            cond: {
+              $and: [
+                {
+                  $gte: ['$$ws.dueDate', fromDate],
+                }, {
+                  $lte: ['$$ws.dueDate', toDate],
+                },
+              ],
+            },
+          },
+        },
+        weeklySummariesCount: 1,
+      },
+    },
+    ]);
+  };
+
+  /**
+   * Checks whether a date belongs to a specific week based on week index.
+   *
+   * @param {string} dueDate The date to check.
+   * @param {integer} weekIndex An integer in the range 0 - 2, where 0 (this week),
+   *                      1 (last week) or 2 (the week before last).
+   * @return {boolean} True if match, false otherwise.
+   */
+  const doesDateBelongToWeek = function (dueDate, weekIndex) {
+    const pstStartOfWeek = moment().tz('America/Los_Angeles').startOf('week').subtract(weekIndex, 'week');
```

Comment:
Should we be setting the time zone for LA? Are we just going off California time, for now, just unsure if this should be hardcoded here? Might be a good idea to have a ‘const timeZone’ holding the time zone that we can alter later?

Reply:
Good question and yes it should be set to LA, I had that discussion with Jae already at some point and he wants it all LA time, so we could have it hard coded like this.